src/signing/ledger.rs

```diff
@@ -60,27 +60,24 @@ impl super::Signer for LedgerNanoSigner {
             true => iota_ledger::TransportTypes::TCP,
             false => iota_ledger::TransportTypes::NativeHID,
         };
-        match iota_ledger::get_opened_app(&transport_type) {
-            Ok((name, version)) => LedgerStatus {
-                connected: true,
-                locked: false,
-                app: Some(crate::LedgerApp { name, version }),
-            },
-            Err(_) => {
-                log::info!("get_ledger");
-                let (connected, locked) =
-                    match iota_ledger::get_ledger(crate::signing::ledger::HARDENED, is_simulator).map_err(Into::into) {
-                        Ok(_) => (true, false),
-                        Err(crate::Error::LedgerDongleLocked) => (true, true),
-                        Err(_) => (false, false),
-                    };
-                LedgerStatus {
-                    connected,
-                    locked,
-                    app: None,
-                }
-            }
-        }
+
+        let app = match iota_ledger::get_opened_app(&transport_type) {
+            Ok((name, version)) => Some(crate::LedgerApp { name, version }),
+            _ => None,
+        };
+
+        log::info!("get_ledger");
+        let (connected_, locked) =
+            match iota_ledger::get_ledger(crate::signing::ledger::HARDENED, is_simulator).map_err(Into::into) {
+                Ok(_) => (true, false),
+                Err(crate::Error::LedgerDongleLocked) => (true, true),
```

Comment:
I think `get_ledger` never returns `LedgerDongleLocked`. There are commands that are allowed although the Ledger is connected. But there are commands that could be executed afterwards to figure out if it's locked. For instance the `get_data_buffer_state` call that always can be executed is only allowed if the ledger is not locked. I think, that would be a solid test.

Reply:
This call should return if the ledger is locked or not:

https://github.com/iotaledger/ledger.rs/commit/4d0656cedb744686fdb342ede15ce47ec92c532d#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R271